var/spack/repos/builtin/packages/root/package.py

```diff
@@ -50,71 +61,421 @@ class Root(CMakePackage):
+    # variant('mysql', default=False) - not supported by spack
+    variant('odbc', default=False,
+        description='Enable ODBC support')
+    variant('opengl', default=True,
+        description='Enable OpenGL support')
+    # variant('oracle', default=False) - not supported by spack
+    variant('postgres', default=False,
+        description='Enable postgres support')
+    variant('pythia6', default=False,
+        description='Enable pythia6 support')
+    variant('pythia8', default=False,
+        description='Enable pythia8 support')
+    variant('python', default=True,
+        description='Enable Python ROOT bindings')
+    variant('qt4', default=False,
+        description='Enable Qt graphics backend')
+    variant('r', default=False,
+        description='Enable R ROOT bindings')
+    variant('rpath', default=False,
+        description='Enable RPATH')
+    variant('rootfit', default=True,
+        description='Build the libRooFit advanced fitting package')
+    variant('root7', default=False,
+        description='Enable ROOT 7 support')
+    variant('shadow', default=False,
+        description='Enable shadow password support')
+    variant('sqlite', default=False,
+        description='Enable SQLite support')
+    variant('ssl', default=False,
+        description='Enable SSL encryption support')
+    variant('table', default=False,
+        description='Build libTable contrib library')
+    variant('tbb', default=True,
+        description='TBB multi-threading support')
+    variant('test', default=False,
+        description='Enable test suit of ROOT with CTest')
+    variant('threads', default=True,
+        description='Enable using thread library')
+    variant('tiff', default=True,
+        description='Include Tiff support in image processing')
+    variant('tvma', default=True,
+        description='Build TMVA multi variate analysis library')
+    variant('unuran', default=True,
+        description='Use UNURAN for random number generation')
+    variant('vc', default=False,
+        description='Enable Vc for adding new types for SIMD programming')
+    variant('vdt', default=True,
+        description='Enable set of fast and vectorisable math functions')
+    variant('x', default=True,
+        description='Enable set of graphical options')
+    # variant('xinetd', default=False,
+    #    description='Enable a daemon process manager')
+    variant('xml', default=True,
+        description='Enable XML parser interface')
+    variant('xrootd', default=False,
+        description='Build xrootd file server and its client')
 
-    depends_on('binutils')
-    depends_on('zlib')
-    # depends_on('unuran')
-    depends_on('freetype')
-    depends_on('pcre')
+    # ################# Variants dependencies ##########################
+
+    # TODO
+    # Davix variant also requires openssl support
+    depends_on('openssl', when='+davix')
+
+    # TODO
+    # TMVA variant also requires
```

Comment:
@adamjstewart what's the best way to define dependencies between variants while keeping them as individual variants?
Example: `+tmva` will explicitly require `+gsl +math` (but I would like to avoid having `tmva, gsl, math` enabled by a single variant)

Reply:
@JavierCVilla One thing I've seen is _e.g._ `conflicts('+tmva', '~gsl')`, which of course requires the user to specify both explicitly, but at least complains when they don't.